opm/simulators/timestepping/ConvergenceReport.hpp

```diff
@@ -53,6 +53,31 @@ namespace Opm
             NotANumber = 3,
         };
 
+        struct PenaltyCard {
+            int nonConverged{0};
+            int distanceDecay{0};
+            int largeWellResiduals{0};
+
+            int total() const {
+                return nonConverged + distanceDecay + largeWellResiduals;
+            }
+
+            PenaltyCard& operator+=(const PenaltyCard& other) {
+                nonConverged += other.nonConverged;
+                distanceDecay += other.distanceDecay;
+                largeWellResiduals += other.largeWellResiduals;
+                return *this;
+            }
+
+            template <typename Serializer>
+            void serializeOp(Serializer& serializer)
```

Comment:
maybe this will change, but I don't see any situation where this needs to be serialized, obviously not in the initial eclipse state broadcast, and as restart-serialization happens at report steps, there is also no reason to serialize as afaict this is reset at each nonlinear iteration, right?

Reply:
You are perhaps correct. I just followed the class structure of the other classes but was unsure if I actually needed this part. Subject to change, I am planning that for each report there is a penalty card. Such that at the end of the simulation, I can add the counts to the INFORITER (or similar) for analysis. Do I need the serializer for that?

Reply:
Just picking up this point now, sorry for the delayed response.

> I don't see any situation where this needs to be serialized,

I switched to using the `serializeOp()` protocol for the `ConvergenceReport` object in commit 0b40277e0 (PR #5338).  As long as this information is intended for output to the `.INFOITER` file, it needs to have a `serializeOp()` member function.